### PR TITLE
Feat/registration

### PR DESCRIPTION
**Pull request by ninadarbaidze**

### CODE REVIEW

File: components/Landing/Signup/Signup.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import React, { useContext } from 'react';
+import { Modal, Backdrop, Input, Button } from 'components';
+import { MovieQuotesContext } from 'store';
+import { useTranslation } from 'react-i18next';
+import { useSignup, GoogleBtn } from 'components';
+
+const SignUp = () => {
```
Comment (GigaGiorgadze) [inline components/Landing/Signup/Signup.tsx]: move useContextand useTranslation to custom hook useSignup

File: components/Landing/Signup/useSignup.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { useContext } from 'react';
+import { useFormik } from 'formik';
+import { RegistrationSchema } from 'schema';
+import { signup } from 'services';
+import { useRouter } from 'next/router';
+import { MovieQuotesContext } from 'store';
+
+export const useSignup = () => {
```
Comment (GigaGiorgadze) [inline components/Landing/Signup/useSignup.ts]: sign up isn't one word its supposed to be useSignUp

File: components/shared/GoogleBtn/GoogleBtn.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { GoogleIcon } from 'public/assets/svgs';
+import { signIn } from 'next-auth/react';
+
+const GoogleBtn = () => {
+  const submitHandler = () => {
```
Comment (GigaGiorgadze) [inline components/shared/GoogleBtn/GoogleBtn.tsx]: this goes to helpers

File: components/shared/Header/Header.tsx [hunk 4]
```diff
@@ -1,10 +1,12 @@
-import React from 'react';
+import { useContext } from 'react';
 import { Button } from 'components';
 import { LanguageSwitchBtn } from 'components';
 import { useTranslation } from 'react-i18next';
+import { MovieQuotesContext } from 'store';
 
 const Header = () => {
   const { t } = useTranslation();
+  const ctx = useContext(MovieQuotesContext);
```
Comment (GigaGiorgadze) [inline components/shared/Header/Header.tsx]: to cutsom hook this goes,

File: components/shared/Modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { ModalTypes } from './types';
+
+const Modal: React.FC<ModalTypes> = (props) => {
```
Comment (GigaGiorgadze) [inline components/shared/Modal/Modal.tsx:4]: u disctructure props in other files, but not here, pick which one u want to do

File: public/assets/svgs/GoogleIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { ClassName } from 'types';
+
+const GoogleIcon: React.FC<ClassName> = (props) => {
+  return (
+    <svg
+      width='16'
```
Comment (GigaGiorgadze) [inline public/assets/svgs/GoogleIcon.tsx]: why are icons in public? move them to components/icons

File: schema/RegistrationSchema.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import * as Yup from 'yup';
+
+const RegistrationSchema = Yup.object({
```
Comment (GigaGiorgadze) [inline schema/RegistrationSchema.ts]: both this and file name in camelCase

File: store/MovieQuotesContext.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import { Children, ContextData } from 'types';
+
+export const MovieQuotesContext = createContext({
+  authModalState: false,
+  changeRegistrationModalState: () => {},
+  loginModalState: false,
+  changeLoginModalState: () => {},
+});
+
+export const MovieQuotesContextProvider: React.FC<Children> = (props) => {
```
Comment (GigaGiorgadze) [inline store/MovieQuotesContext.tsx]: i don't think name of this file is appropriate, movieQuotesContext and u are storing Login and register state here, change name

File: .env [hunk 9]
```diff
@@ -0,0 +1,9 @@
+NEXT_PUBLIC_API_URL=http://localhost:3001
+
+
+NEXT_PUBLIC_GOOGLE_ID=674266662916-maqgnm9m8f9jgffft28khekrtkue9tti.apps.googleusercontent.com
```
Comment (GigaGiorgadze) [inline .env]: and now i know your google ID :eye: :lips: :eye:  , remove .env from git

**Review by GigaGiorgadze — CHANGES_REQUESTED** [review]

File: components/shared/GoogleBtn/GoogleBtn.tsx (hunk 3 above)
Comment (ninadarbaidze) [reply components/shared/GoogleBtn/GoogleBtn.tsx]: Google button component or submitHandler?

Comment (GigaGiorgadze) [reply components/shared/GoogleBtn/GoogleBtn.tsx]: submitHandler

File: schema/RegistrationSchema.ts (hunk 7 above)
Comment (GigaGiorgadze) [reply schema/RegistrationSchema.ts]: file name is still in PascalCase

File: store/MovieQuotesContext.tsx (hunk 8 above)
Comment (GigaGiorgadze) [reply store/MovieQuotesContext.tsx]: it still has same name, chane it

**Review by GigaGiorgadze — CHANGES_REQUESTED** [review]
check converstations i unresolved

File: store/MovieQuotesContext.tsx (hunk 8 above)
Comment (ninadarbaidze) [reply store/MovieQuotesContext.tsx]: I change it, but apperantly there's some problem in git, I'll check it

File: schema/RegistrationSchema.ts (hunk 7 above)
Comment (ninadarbaidze) [reply schema/RegistrationSchema.ts]: I had to clear git cach

**Review by GigaGiorgadze — APPROVED** [review]

**Pull request merged**